services/frontend-service/src/ui/AppBar/ActionsCart.tsx

```diff
@@ -70,6 +70,15 @@ const ActionsList = (props: { openNotification: (msg: string) => void }) => {
     }, [openNotification]);
     const [doActions, doActionsState] = callbacks.useBatch(actions, actionsSucceeded, actionsFailed);
 
+    useEffect(() => {
+        window.onbeforeunload = (e: BeforeUnloadEvent) => {
```

Comment:
Isn’t this triggered every time? Shouldn’t be only when the cart not empty ?

Reply:
Yes!! :sweat_smile: 
I changed it.